keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components/README.md

```diff
@@ -124,10 +124,11 @@ best practices.
 
 ## Proposal
 
-I propose to remove klog specific feature flags in Kubernetes core components
-(kube-apiserver, kube-scheduler, kube-controller-manager, kubelet) and leave
-them with defaults. From klog flags we would remove all flags besides "-v"
-and "-vmodule".
+We propose to remove klog specific feature flags in Kubernetes components
+(including, but not limited to, kube-apiserver, kube-scheduler,
```

Comment:
Just to confirm, this extends the previous list of components to projects owned by K8s Org. In more detail, components that use kubernetes/component-base/logs, which also include kube-proxy, cluster-autoscaler etc.

Reply:
Everything that uses `kubernetes/component-base/logs` to add klog flags worked without modifications, so sig-instrumentation doesn't need to do additional work because of this increased scoped.

Moving some options into `LoggingConfiguration` is more intrusive, but only a few commands use that, so it's still doable.

Reply:
Important thing to note `kubernetes/component-base/logs` is used by non-Kubernetes Org projects, we didn't give any compatibility promises, but I expect some people will not be happy. Possibly this could be solved with proper documentation that would describe how those project could directly use klog.

Reply:
> Important thing to note `kubernetes/component-base/logs` is used by non-Kubernetes Org projects, we didn't give any compatibility promises, but I expect some people will not be happy. Possibly this could be solved with proper documentation that would describe how those project could directly use klog.

Would it make sense to add something about this to alpha or beta graduation criteria? All we have is `Documentation on migrating off klog flags is publicly available` but that doesnt seem to cover @serathius suggestion above.

Reply:
Where would such documentation live? A blog post describing the recent changes?

There's the API documentation in gopkg. Do we need more than that?

Reply:
Yeah this would need to go on the blog at minimum. Ensure we are engaging the comms team.

Reply:
I can write a blog post and submit a talk for the next KubeCon.